htdocs/xoops_lib/Xoops/Core/Security.php

```diff
@@ -189,11 +189,12 @@ public function checkBadips()
             && $_SERVER['REMOTE_ADDR'] != ''
         ) {
             foreach ($xoops->getConfig('bad_ips') as $bi) {
-                if (!empty($bi) && preg_match('/' . $bi . '/', $_SERVER['REMOTE_ADDR'])) {
-                    exit();
+                if (!empty($bi) && false !== strpos($_SERVER['REMOTE_ADDR'],$bi)) {
```

Comment:
The change in the matching would create false positives. I.e. a ban on '2.3.4.5' would ban '12.3.4.5', '252.3.4.5', etc.